modin/pandas/series.py

```diff
@@ -926,11 +926,21 @@ def equals(self, other):  # noqa: PR01, RT01, D200
         """
         Test whether two objects contain the same elements.
         """
-        return (
-            self.name == other.name
-            and self.index.equals(other.index)
-            and self.eq(other).all()
```

Comment:
`equals` shouldn't be implemented through `eq`; in general, their behavior is different.